setup.py

```diff
@@ -1,11 +1,44 @@
 import glob
+import pathlib
 import sys
 
 from setuptools import find_packages, setup
+from setuptools.command.develop import develop
 
 if sys.version_info < (3, 6):
     sys.exit("Please use Python version 3.6 or higher.")
 
+
+# Handle protobuf
+class CustomDevelopCommand(develop):
+    def run(self):
+        # we need to import this here or else grpc_tools would have to be
+        # installed in the system before we could run the setup.py
+        import numproto
+        from grpc_tools import protoc
+
+        develop.run(self)
+
+        # get the path of the numproto protofiles
+        # this will give us the path to the site-packages where numproto is
+        # installed
+        numproto_path = pathlib.Path(numproto.__path__[0]).parent
+
+        proto_files = glob.glob("./protobuf/xain/grpc/*.proto")
+        command = [
+            "grpc_tools.protoc",
+            f"--proto_path={numproto_path}",
+            "--proto_path=./protobuf",
+            "--python_out=./",
+            "--grpc_python_out=./",
+            "--mypy_out=./",
+        ] + proto_files
+
+        print("Building proto_files {}".format(proto_files))
```

Comment:
As we require Python 3.6, we could use f-strings: `print(f"Building proto_files {proto_files}")`.

Reply:
True. Currently we have both going on in the code.

At some point we should agree on a style guide.